public/app/ui/components/search/basic/BasicSearchComponent.js

```diff
@@ -0,0 +1,111 @@
+                    getFieldAndTextForQuery = function (query) {
+                        var selectedKey, selectedField;
+                        if (!query || !query.children || query.children.length === 0) {
+                            return undefined;
+                        }
+                        selectedKey = query.children[0].field;
+                        if (!selectedKey) {
+                            return undefined;
+                        }
+                        selectedField = _.find(parameters.fields(), { key: selectedKey });
+                        if (!selectedField || !selectedField.basicSearch) {
+                            return undefined;
+                        }
+                        return {
+                            field: selectedKey,
+                            text: _(query.children).filter({ field: selectedKey }).map('value').join(' ')
+                        };
+                    },
+                    getQueryForFieldAndText = function (field, text) {
+                        return text.length === 0 ? undefined : {
+                            operator: 'AND',
+                            children: _.map(
+                                text.split(/\s+/),
+                                function (term) {
+                                    return {
+                                        field: field,
+                                        comparator: 'contains',
+                                        value: term
+                                    };
+                                }
+                            )
+                        };
+                    };
+
+                if (!parameters.fields || parameters.fields().length === 0) {
+                    throw new Error('BasicSearchComponent missing required parameter: `fields`.');
+                }
+                if (!parameters.queryObservable || !ko.isObservable(parameters.queryObservable)) {
+                    throw new Error('BasicSearchComponent missing or invalid required parameter: `queryObservable`.');
+                }
+
+                initialFieldAndText = getFieldAndTextForQuery(parameters.queryObservable());
+
+                selectedSearchFieldKey = ko.observable(initialFieldAndText && initialFieldAndText.field ? initialFieldAndText.field : parameters.fields()[0].key);
+                selectedSearchFieldKey.subscribe(function (key) {
+                    parameters.queryObservable(getQueryForFieldAndText(key, this.searchText()));
+                }.bind(this));
+
+                this.searchText = ko.observable(initialFieldAndText ? initialFieldAndText.text : '');
+                this.searchText.subscribe(function (text) {
+                    parameters.queryObservable(getQueryForFieldAndText(selectedSearchFieldKey(), text));
+                }.bind(this));
+
+                parameters.queryObservable.subscribe(function (query) {
```

Comment:
When the query changes, update the values in this object.  This is what copies changes from the advanced search into the basic search.